### ISSUE TITLE
Dump command error [Windows]

### ISSUE THREAD

**Issue by homeedog** [kakaroto/ps3xport#5]
When using the dump command on Windows the binary returns“Error making directories” and creates an empty folder where the freshly decrypted .dat files should be placed.

**Comment by RickyVonc** [comment]
try in C:\ , it work for me

**Comment by homeedog** [comment]
Thanks, that did work. Kakaroto should definitely add that you can only Dump to C:\ in his post. (I'm not sure if this is a bug, or just not mentioned...)

**Comment by tolga9009** [comment]
This is probably a bug and not a feature. By design, it should work for any path. mkdir_recursive() in tools.c:457 (leading to this issue on fail) is strictly designed for UNIX operating system. Why it's working on C:\ root directory? No idea, I'm not experienced in Windows-specific programming - but it's a cool find!

**Comment by homeedog** [comment]
How about the CreateBackup command? It creates the backup but the files created are corrupted... I can't tell whether or not it is the packaging of the backup .dat files or the unreliable Dump corrupting the files from the beginning...

**Comment by kakaroto** [comment]
Good thing it works for C: at least! I have no idea why it doesn't work for other paths. I hope someone can fix it and send me the patches :)

**Comment by homeedog** [comment]
Weird huh? There is also something going on with the CreateBackup command... Data is finding itself put in other archives! I will have to investigat e this further...

**Comment by kakaroto** [comment]
Not sure what you mean by that, but please do investigate further and open issues as you see fit.
Unfortunately, I don't have time (or motivation) to be maintaining ps3xport, I just released it as is (3 years old code) without any recent testing so anyone is free to look over the code and fix stuff and I'll be happy to merge any pull requests that come my way.

**Comment by kakaroto** [comment]
By the way, @homeedog, make sure you pull the changes from version 0.2 which had a fix for data corruption when using ExtractFile, maybe that's why you had corrupted files ?

**Comment by homeedog** [comment]
You've done more than enough m8. You've given an old OFW user hope! In all seriousness I will investigate further and get back with any new info.

**Comment by homeedog** [comment]
I did get 0.2 win binary. I'm also thinking up a way to actually delete files from backups since it seems easier to do than fix the createbackup. by fixing either one you could essentially perform the same tasks by just deleting the root directory on a backup and add a new one in using Add/Add protected... But thanks for the tool and the work put in and enjoy retirement!

**Comment by spewfr** [comment]
I can get most everything working by just dumping to C:\ (or external D:\). So really, just always work with the root of the drive, and you should be good to go! I'm finding this is a very useful tool.
Thanks again kakaroto! :)

**Comment by homeedog** [comment]
Can you build correctly off C:/? That's my biggest issue right now...

**Comment by spewfr** [comment]
I can extract a folder, a file, and re-install a folder and file all from
C. I don't know about dumping.

**Comment by homeedog** [comment]
I would like to dump my backup to C:, which works. Then edit (delete &add files) and use CreateBackup to repackage them... But create backup always skrews up my backup during restore and I can see that the file sizes are totally off...

**Comment by spewfr** [comment]
Yeah no, composing a backup is broken. I don't even understand how to add
the right folders. I just tried installing a backup with a modified file
structure after using AddProtected - with no dice. I'll try again.

**Comment by spewfr** [comment]
Another weird occurrence I just ran into; I got an error while trying to create a fresh backup: "The backup operation could not be completed. (some error code)". I tried it again, changing nothing, and I'm not backing up again. Weird O.o

**Comment by spewfr** [comment]
I tried AddProtected, and successfully added the directory/file, but I tried to extractfile the filrs that I added and got some jumbled mess. I think that not maintaining continuity in the archive and index folder structure (i.e. you have a backup with two folders, folders 1 and 2, and then adding another folder called 1 after; "1,2,1" is not a good way to go.) Whenever you add a new folder, the folder gets added to the end of the index. This might be confusin to the PS3/ps3xport app. I'll continue screwing around with it, but I cannot code in C (or whatever language this is made in) for my life! Hopefully somebody else can fix these glitches.

**Comment by spewfr** [comment]
I don't know what the rule is about double-posting here, but... I've successfully extracted, edited, and re-archived a backup of mine, but when I go to restore the backup on the PS3, I get an error saying "this backup was created on a different system," and then the system can't even put the generic data back on the hard drive (Error 80010037). Something is going wrong parsing(?) the data that's put back into the backup files when you re-compile.
I really hope somebody pulls this under their wing, because I sure as heck don't know much about coding something this advanced. Maybe it's time I learned?

**Comment by windrider42** [comment]
I know I got the “Error making directories”  when trying to use G:/ as a directory
Otherwise great job KaKaRoTo on your release. I fully understand about life and not having time for all this console stuff,
Will use C:/ or try D:/
